packages/core/src/types/data/common.ts

```diff
@@ -44,11 +47,87 @@ export interface Sample {
 /**
  * The data loader to used to load dataset.
  */
-export interface DataLoader {
-  len: () => Promise<number>;
-  getItem: (id: number) => Promise<Sample>;
-  next?: () => Promise<Sample>;
-  nextBatch?: (batchSize: number) => Promise<Array<Sample>>;
+export abstract class DataLoader{
```

Comment:
```suggestion
export abstract class DataLoader {
```

Reply:
And a unit tests for this class is required :)

Reply:
Lint problem has been fixed. I will add a unit test ASAP

Reply:
Unit test has been added. Please have a check pls :)